test/reducers/campaigns.spec.js

```diff
@@ -0,0 +1,95 @@
+import deepFreeze from 'deep-freeze';
+import campaigns from '../../app/assets/javascripts/reducers/campaigns';
+import {
+    RECEIVE_COURSE_CAMPAIGNS,
+    RECEIVE_ALL_CAMPAIGNS,
+    ADD_CAMPAIGN,
+    SORT_CAMPAIGNS_WITH_STATS,
+    SORT_ALL_CAMPAIGNS } from '../../app/assets/javascripts/constants';
+
+const campaignsArray = [
+      { id: 2, slug: 'second_campaign', description: 'second description' },
+        { id: 1, slug: 'first_campaign', description: 'first description' }
+      ];
+ const sortedcampaignsArray = [
+      { id: 1, slug: 'first_campaign', description: 'first description' },
+      { id: 2, slug: 'second_campaign', description: 'second description' },
+      ];
+
+describe('campaigns reducer', () => {
+    test('should return initial state when no action nor state is provided', () => {
+        const newState = campaigns(undefined, { type: null });
+        expect(newState.campaigns).toEqual([]);
+        expect(newState.all_campaigns).toEqual([]);
+        expect(newState.isLoaded).toBe(false);
+    });
+
+    test('should and and return campaigns data with ADD_CAMPAIGN ans set isLoaded to true', () => {
+        const initialState = { campaigns: [], all_campaigns: [], isLoaded: false };
+        deepFreeze(initialState);
+
+        const mockedAction = {
+          type: ADD_CAMPAIGN,
+          data: { course: { campaigns: campaignsArray } }
```

Comment:
This test doesn't reflect the actual usage of the ADD_CAMPAIGN action, as it is only used to add a single campaign per action, but this test goes from zero to two campaigns. (This is still theoretically possible, as another campaign could have been added independently, without this browser store knowing about it, so that adding one campaign would result in a response that includes more than one new campaign. This test can be kept, but it ought to have a comment explaining that sort of situation.)